packages/hardhat-core/test/internal/core/config/config-validation.ts

```diff
@@ -449,6 +449,66 @@ describe("Config validation", function () {
             ERRORS.GENERAL.INVALID_CONFIG
           );
 
+          expectHardhatError(
+            () =>
+              validateConfig({
+                networks: {
+                  [HARDHAT_NETWORK_NAME]: {
+                    accounts: [{ privateKey: "0xxxxx", balance: 213 }],
+                  },
+                },
+              }),
+            ERRORS.GENERAL.INVALID_CONFIG
+          );
+
+          expectHardhatError(
+            () =>
+              validateConfig({
+                networks: {
+                  [HARDHAT_NETWORK_NAME]: {
+                    accounts: [{ privateKey: "0xxxxx", balance: "0.1231" }],
+                  },
+                },
+              }),
+            ERRORS.GENERAL.INVALID_CONFIG
+          );
+
+          expectHardhatError(
+            () =>
+              validateConfig({
+                networks: {
+                  [HARDHAT_NETWORK_NAME]: {
+                    accounts: [{ privateKey: "0xxxxx", balance: "001231" }],
+                  },
+                },
+              }),
+            ERRORS.GENERAL.INVALID_CONFIG
+          );
+
+          expectHardhatError(
+            () =>
+              validateConfig({
+                networks: {
+                  [HARDHAT_NETWORK_NAME]: {
+                    accounts: [{ privateKey: "0xxxxx", balance: ".02123" }],
+                  },
+                },
+              }),
+            ERRORS.GENERAL.INVALID_CONFIG
+          );
+
+          expectHardhatError(
+            () =>
+              validateConfig({
+                networks: {
+                  [HARDHAT_NETWORK_NAME]: {
+                    accounts: [{ privateKey: "0xxxxx", balance: "-123" }],
+                  },
+                },
+              }),
+            ERRORS.GENERAL.INVALID_CONFIG
+          );
```

Comment:
There should be a couple of tests for _valid_ configs, that would've watched the regex error. For example:

- 0
- 1
- 10
- 1234